### PR TITLE
fix(helm): process  dependencies import-values

### LINKED ISSUE #9658: helm lint not finding subchart imported values when templates in main chart
While working on some complex helm charts I isolated a weird lint issue. I worked on isolating what triggered it and wrote a sample reproducer:

[helm-lint-issue-reproducer.zip](https://github.com/helm/helm/files/6923230/helm-lint-issue-reproducer.zip)

The following setup:

* `libchart` is a library chart defining a template and a value copied.value
* `subchart` depends on `libchart` and uses the template with `{{ template name . }}`
* `mainchart` is kind of an umbrella chart that depends on `subchart`, but defines a YAML file too

Linting on the mainchart after dependency builds are done breaks with the following error:

```
[ERROR] templates/: template: mainchart/charts/subchart/charts/libchart/templates/_object.yaml:5:19: executing "object.template" at <.Values.copied.value>: nil pointer evaluating interface {}.value
```

If you remove the templates folder from the mainchart, lint starts working again.

Output of `helm version`:

```
version.BuildInfo{Version:"v3.5.4", GitCommit:"1b5edb69df3d3a08df77c9902dc17af864ff05d1", GitTreeState:"clean", GoVersion:"go1.15.11"}
```

This issue is independent of the version of kubectl or the cloud provider chosen: it is pure helm lint.

### PR DESCRIPTION
**Pull request by stuartdrennan**
When running helm lint, import-values for dependencies are ignored,
also added test for linting chart with import-values

Closes #9658

Signed-off-by: Stuart Drennan <stuart.drennan@gmail.com>

**What this PR does / why we need it**:

When running helm lint, import-values for dependencies are ignored this fixes that by processing dependencies as part of template rule, also added test for linting chart with import-values

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by bacongobbler — APPROVED** [review]
code LGTM. Sorry for letting this fall through the cracks.

Will need another maintainer's approval before merging as per the LGTM policy

**Comment by stuartdrennan** [comment]
No worries @bacongobbler, thanks for jumping on it

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @stuartdrennan

**Pull request merged**
